warehouse/packaging/tasks.py

```diff
@@ -508,3 +509,30 @@ def populate_data_using_schema(file):
                 json_rows, table_name, job_config=LoadJobConfig(schema=table_schema)
             ).result()
             break
+
+
+@tasks.task(ignore_result=True, acks_late=True)
+def send_pep_715_notices(request):
+    """
+    Notifies projects that have uploaded eggs since Jan 1, 2023 of PEP 715
+    """
+    projects = set()
+    for release_file in request.db.query(File).filter(
+        File.packagetype == "bdist_egg",
+        File.upload_time >= "2023-01-01",
+    ):
+        projects.add(release_file.release.project)
+
+    for project in projects:
+        contributors = project.users
+        if project.organization:
+            contributors += project.organization.owners
+            for teamrole in project.team_project_roles:
+                contributors += teamrole.team.members
+
+        for contributor in sorted(set(contributors)):
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**thought:** I've seen this kind of pattern a couple of times now, so we likely want to queue up a feature request for `Project.all_members` or such that figures this out.

Reply:
I agree, I think `Project.contributors` would fit nicely since thats how it is used most frequently.

Also an update to `User.projects` would be a step towards https://github.com/pypi/warehouse/issues/13840

Reply:
I'd suggest both of those are addressed in a separate PR. (This use case is temporary).

Reply:
Of course! I use [Conventional Comments](https://conventionalcomments.org/#:~:text=thought%3A,and%20mentoring%20opportunities.) for code reviews.